applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c

```diff
@@ -361,14 +303,16 @@ static void base_recv_cb(struct bt_bap_broadcast_sink *sink, const struct bt_bap
 static void syncable_cb(struct bt_bap_broadcast_sink *sink, bool encrypted)
 {
 	int ret;
-	static uint8_t bis_encryption_key[16];
+	static uint8_t bis_encryption_key[BT_ISO_BROADCAST_CODE_SIZE] = {0};
 	struct bt_bap_stream *audio_streams_p[] = {&audio_streams[active_stream_index]};
 
 	LOG_DBG("Broadcast sink is syncable");
 
-#if (CONFIG_BT_AUDIO_BROADCAST_ENCRYPTED)
-	strncpy(bis_encryption_key, CONFIG_BT_AUDIO_BROADCAST_ENCRYPTION_KEY, 16);
-#endif /* (CONFIG_BT_AUDIO_BROADCAST_ENCRYPTED) */
+	if (IS_ENABLED(CONFIG_BT_AUDIO_BROADCAST_ENCRYPTED)) {
+		memcpy(bis_encryption_key, CONFIG_BT_AUDIO_BROADCAST_ENCRYPTION_KEY,
+		       MIN(strlen(CONFIG_BT_AUDIO_BROADCAST_ENCRYPTION_KEY),
+			   ARRAY_SIZE(bis_encryption_key)));
```

Comment:
This does an implicit truncate of the `CONFIG_BT_AUDIO_BROADCAST_ENCRYPTION_KEY`. That seems like a poor user experience. Suggest to have a `BUILD_ASSERT` ensuring that `CONFIG_BT_AUDIO_BROADCAST_ENCRYPTION_KEY` is not more than 16 octets